### PR TITLE
Etching out of what n-makefile.json could be

### PR DESCRIPTION
**Pull request by matthew-andrews**
- Combines the two webpack config files.
- Removes the requirement for people to have `node_modules/.bin` on their paths.
- Defines the entry point for Sass + JS in a config file.

@ironsidevsquincy

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -3,15 +3,12 @@ const autoprefixer = require('autoprefixer');
 const BowerWebpackPlugin = require('bower-webpack-plugin');
```
Comment (matthew-andrews) [inline webpack.config.js]: I'll make a second PR later that will pull this file out and put it into the `n-makefile` and then try it out in another app :smile:

File: webpack.config.js [hunk 2]
```diff
@@ -30,7 +23,7 @@ module.exports = {
 				],
 				query: {
 					cacheDirectory: true,
-					presets: ['react', 'es2015'],
+					presets: (require('./package.json').dependencies.react ? ['react', 'es2015'] : ['es2015']),
```
Comment (matthew-andrews) [inline webpack.config.js]: for other apps mostly

File: webpack.config.js [hunk 3]
```diff
@@ -3,19 +3,12 @@ const autoprefixer = require('autoprefixer');
...
-	entry: {
-		'main.js': './main.js',
-		'main.css': './main.scss',
-		'ie8.css': './ie8.scss'
-	},
-	output: {
-		path: './public',
-		filename: '[name]'
-	},
+	devtool: process.argv.indexOf('--dev') === -1 ? 'source-map' : 'cheap-module-eval-source-map',
```
Comment (ironsidevsquincy) [inline webpack.config.js]: we've had a few issues with using `cheap-module-eval-source-map`, i.e. breakpoints don't work probably. might be worth just using `source-map` for all - though will have a performance impact. or maybe one of the other options

https://webpack.github.io/docs/configuration.html#devtool

File: webpack.config.js [hunk 4]
```diff
@@ -3,15 +3,12 @@ const autoprefixer = require('autoprefixer');
...
 module.exports = {
-	devtool: 'source-map',
+	devtool: process.argv.indexOf('--dev') === -1 ? 'source-map' : 'cheap-module-eval-source-map',
 	context: path.join(__dirname, 'client'),
-	entry: {
-		'main.js': './main.js',
-		'main.css': './main.scss',
-		'ie8.css': './ie8.scss'
-	},
+	entry: config.assets.entry,
```
Comment (ironsidevsquincy) [inline webpack.config.js]: so simple - that's cool

**Comment by ironsidevsquincy** [comment]
:+1:

File: webpack.config.js (hunk 3 above)
Comment (matthew-andrews) [inline webpack.config.js]: ok, i'll do that

**Pull request merged**